### ISSUE TITLE
Missing epilogue?

### ISSUE THREAD

**Issue by emilaxelsson** [Feldspar/feldspar-compiler#201]
The epilogue is discarded from the generated code [here](https://github.com/Feldspar/feldspar-compiler/blob/7504ed9a2c8c12438d8f64c798318d1010829a30/src/Feldspar/Compiler/Imperative/FromCore.hs#L272). `confiscateBigBlock` resets the epilogue in the `CodeWriter` so the epilogue is only present in the result of the bind, but there it is ignored (the second wildcard).

(Note: I have just changed the type of `confiscateBigBlock` a bit but the code is equivalent to how it was before.)

I'm a bit lost in the code generation. Where should the epilogue go?

@pjonsson, @emwap

**Comment by emilaxelsson** [comment]
OK, thanks. I guess the right thing to do is to make some examples (or maybe there already are some) and check the generated code. I'll leave the issue open until I have time to do this.

**Comment by pjonsson** [comment]
Yes, a future that contains something that allocates memory (`sequential` perhaps?) that is not immediately written into the result array is my guess. My second guess is a future that has a local future inside its body and waits for that future before returning.
